src/app/shared/services/user-session.service.ts

```diff
@@ -29,18 +30,18 @@ export class UserSessionService implements OnDestroy {
 
   constructor(
     private authService: AuthService,
-    private http: CurrentUserHttpService,
+    private currentUserService: CurrentUserHttpService,
   ) {
-    this.subscription = this.authService.status$.pipe(
-      switchMap(auth => {
-        if (!auth.authenticated) return of<UserProfile|undefined>(undefined);
-        return this.http.getProfileInfo().pipe(
-          catchError(_e => EMPTY)
-        );
-      }),
-      distinctUntilChanged((p1, p2) => p1 === p2 || (!!p1 && !!p2 && p1.groupId === p2.groupId))
+    this.subscription = merge(
+      this.authService.status$.pipe(filter(auth => auth.authenticated)),
+      this.refresh$,
+    ).pipe(
```

Comment:
Instead of the merge, I would have written:
```
this.authService.status$.pipe(
  repeatLatestWhen(this.refresh$),
  switchMap(...
```
(even if I admit the merge works here as you do not need the auth to get the profile)

Reply:
Fixed